### ISSUE TITLE
Error Handling in Lens Library

### ISSUE THREAD

**Issue by torbrenner** [samply/lens#74]
In the PR #67 we had a short discussion about informing the end user about failed queries etc. As i noted there https://github.com/samply/lens/pull/67#discussion_r1531665422, i think the library part of lens should not directly send error messages to the user, as each application could have their own way of handling such events.
So i think lens should offer a way to send such messages to the end application e.g  by emiting through the [event api](https://developer.mozilla.org/en-US/docs/web/api/event) or alternatively by receiving a communication channel from the application.

**Comment by timakro** [comment]
Do we want user-facing errors when something goes wrong that only a developer can fix?

To provide an example: do we want a user-facing error message when [`options` passed to the `lens-options`](https://github.com/samply/lens/blob/main/docs/SETUP.md#configuration) component is invalid?

**Comment by patrick-skowronek** [comment]
Closed by #171

**Comment by timakro** [comment]
@patrickskowronekdkfz I think we want to show error messages potentially in a few more places (every `console.error()` is a potential candidate). That's why I was asking in which cases.

**Comment by patrick-skowronek** [comment]
You are right, it should be at every `console.error()`. Furthermore, there many lines of code where we don't handle error really well. Also I think it would be nice to have a code snippet to see how it works in the documentation.

**Comment by timakro** [comment]
Are you sure we want it in cases that only a developer can fix? See my previous comment https://github.com/samply/lens/issues/74#issuecomment-2594930207

**Comment by patrick-skowronek** [comment]
> Are you sure we want it in cases that only a developer can fix? See my previous comment [#74 (comment)](https://github.com/samply/lens/issues/74#issuecomment-2594930207)

That is a good question. Since the catalog and options are dynamically loaded there might be problem with the connection. Also there could be an issue with the schema. I, personally, would show it since in development you might now look into the console.
